osbs/utils.py

```diff
@@ -333,6 +333,40 @@ def utcnow():
 VALID_BUILD_CONFIG_NAME_CHARS = re.compile('[-a-z0-9]')
 
 
+def sanitize_strings_for_openshift(str1, str2='', limit=64, separator='-'):
+    """
+    OpenShift requires labels and build names to be no more than 64 characters and
+    forbids any characters other than alphanumerics and -.
+
+    Sanitize one or two strings to meet OpenShift's requirements. include an equal number of
+    characters from both strings if the combined length is more than the limit.
+    """
+    str1_san = ''.join(filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(str1)))
+    str2_san = ''.join(filter(VALID_BUILD_CONFIG_NAME_CHARS.match, list(str2)))
+
+    str1_chars = []
+    str2_chars = []
+    groups = ((str1_san, str1_chars), (str2_san, str2_chars))
+
+    size = len(separator)
+    limit = min(limit, 64)
+
+    for i in range(max(len(str1_san), len(str2_san))):
+        for group, group_chars in groups:
+            if i < len(group):
+                group_chars.append(group[i])
+                size += 1
+                if size >= limit:
+                    break
+        else:
+            continue
+        break
+
+    final_str1 = ''.join(str1_chars).strip('-')
+    final_str2 = ''.join(str2_chars).strip('-')
```

Comment:
Wonder if these ought to be `strip(separator)`.

Reply:
Error in the original code, but fixed.